website/static/js/pages/project-dashboard-page.js

```diff
@@ -29,6 +29,27 @@ var node = window.contextVars.node;
 var nodeApiUrl = ctx.node.urls.api;
 var nodeCategories = ctx.nodeCategories || {};
 
+var categoryList = m.prop([]);
+var addProjectTemplate = m.component(AddProject, {
+    buttonTemplate: m('.btn.btn-sm.btn-default[data-toggle="modal"][data-target="#addSubComponent"]', {onclick: function() {
+        $osf.trackClick('project-dashboard', 'add-component', 'open-add-project-modal');
+    }}, 'Add Component'),
+    modalID: 'addSubComponent',
+    title: 'Create new component',
+    parentID: window.contextVars.node.id,
+    parentTitle: window.contextVars.node.title,
+    categoryList: categoryList,
+    stayCallback: function() {
+        // We need to reload because the components list needs to be re-rendered serverside
+        window.location.reload();
+    },
+    trackingCategory: 'project-dashboard',
+    trackingAction: 'add-component',
+    contributors: window.contextVars.node.contributors,
+    currentUserCanEdit: window.contextVars.currentUser.canEdit
+});
+m.mount(document.getElementById('newComponent'), m.component(addProjectTemplate, {wrapperSelector : '#addSubComponent'}));
```

Comment:
It would still be good to do all of this in `document.ready`, in order to avoid trying to mount the component before the document has loaded.